src/app/modules/setting/setting.component.ts

```diff
@@ -8,10 +8,23 @@ import { Router } from '@angular/router';
 })
 export class SettingComponent implements OnInit {
     listopen: boolean;
+    login_time: any;
+    current_time: any;
+    diff: any;
     constructor(private _router: Router) { }
 
     ngOnInit() {
         this.listopen = false;
+        if (localStorage.getItem('token') !== '') {
```

Comment:
we cannot check this in every file ngOnInit 

put this in parent compoonent only. not in every file